src/App.tsx

```diff
@@ -1,223 +1,205 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notification } from './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  deleteTodo,
+  createTodo,
+  updateTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [filter, setFilter] = useState('all');
+  const [subtitleError, setSubtitleError] = useState('');
+  const [title, setTitle] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState(0);
+  const [isLoading, setisLoading] = useState<number[]>([]);
+
+  enum Filter {
+    Active = 'active',
+    Completed = 'completed',
+  }
+  const activeItems = useMemo(() => {
+    return (todos.filter(todo => !todo.completed)).length;
+  }, [todos]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(response => {
+          setTodos(response);
+        })
+        .catch(() => setTimeout(() => {
+          setHasError(true);
+        }, 3000));
+    }
   }, []);
 
+  const visibleTodos = todos.filter(todo => {
+    switch (filter) {
+      case Filter.Completed:
+        return todo.completed;
+      case Filter.Active:
+        return !todo.completed;
+      default:
+        return todo;
+    }
+  });
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setHasError(true);
+      setSubtitleError('Title can\'t be empty');
+
+      return;
+    }
+
+    setisLoading([user?.id || 0]);
+    try {
+      if (user) {
+        const newTodoData = {
+          id: 0,
+          userId: user.id,
+          title,
+          completed: false,
+        };
+
+        setTodos(state => [...state, newTodoData]);
+        const newTodo = await createTodo(user.id, title);
+
+        setTodos(state => [...state, newTodo]
+          .filter(todo => todo.id !== 0));
```

Comment:
we should get todos from server to get relevant data instead of using state. check all cases

Reply:
if I right understood, i fixed right)